### DISCUSSION TITLE
getTooltip async data

### DISCUSSION THREAD

**Discussion by sgavathe** [visgl/deck.gl#5978] [Q&A] [answered]
Does getTooltip support async data? I have a data where it does query based on the attributes. Was wondering, if this is possible>?

`    layer.deck.getTooltip =  async(info: any) => info.object && {    
            html: await this.getTooltip(info, mapView),
            style: {
              backgroundColor: '#f00',
              fontSize: '0.8em'
            } }  `

`  private async getTooltip(item: any, mapView: MapView): Promise<String> `

**Comment by Pessimistress** [comment, marked as answer]
No. You need to implement your own tooltip.

**Reply by sgavathe** [reply]
Thanks. I implemented it. 

**No** means, you can't have a async data inside tooltip function?

**Reply by Pessimistress** [reply]
The built-in `getTooltip` feature does not support async function. See [examples](https://github.com/visgl/deck.gl/blob/7.3-release/docs/developer-guide/interactivity.md#example-display-a-tooltip-for-hovered-object) on how to implement your own.
